include/flang/CMakeLists.txt

```diff
@@ -6,3 +6,7 @@
 #
 #===------------------------------------------------------------------------===#
 
+# Note: Header file install is handled via the lib directory for 
+# now as the files are not separated off from source in the style
+# of clang. 
```

Comment:
I don't understand this. What header files in include/flang need to be installed?

Reply:
It's possible that we just want to install every public header, like llvm and clang does (with _no guarantee of API stability_ as usual)
In that case external tools could use our headers (relative to a _specific version_ of flang of course) in the way that tools like ccls/rtags/etc do for clang

Reply:
I guess I'm confused because there are two different kinds of consumers of flang:
1. Those who want to build and install flang and get a Fortran compiler. They might want a few things like ISO_Fortran_binding.h installed but probably no other .h files.
2. Those who want to build tools using components of flang. The would want the headers and libraries but not things like the driver and predefined module files.

Is there a way to distinguish between these two cases or does everything always get installed?

Reply:
> 
> 
> I guess I'm confused because there are two different kinds of consumers of flang:
> 
>     1. Those who want to build and install flang and get a Fortran compiler. They might want a few things like ISO_Fortran_binding.h installed but probably no other .h files.
> 
>     2. Those who want to build tools using components of flang. The would want the headers and libraries but not things like the driver and predefined module files.
> 
> 
> Is there a way to distinguish between these two cases or does everything always get installed?

At least with the way clang does it, there's no difference in installations as far as I know. Of course, downstream consumers of flang are free to change the cmake file to limit the installed headers to headers that Fortran compiler users want.

Reply:
Yes, David is correct -- I did this purely to match the pattern Clang uses in installing its header files (again to match the community's practice).  There is clearly room to refine and specialize how we want to address header file installs (but that was not my goal with this PR).

Reply:
The headers (and libraries) are installed in different directories for Fortran programmers and llvm programmers.

Reply:
There are suboptions such as LLVM_INSTALL_TOOLCHAIN_ONLY that can be used to install more selectively.